### PR TITLE
Refactor/no space in vectors

### PR DESCRIPTION
**Pull request by macintoshpie**
### Changes
- improved logic for formatting model annotations
- no spaces are inserted after commas
  - implemented b/c Dymola does this inside of vectors (the `{...}`)
  - downside is that now inside of annotations all arguments that are comma separated do not have space after the comma. See the example files. I tried for a few hours to avoid this, but failed to find a good solution. Add this to the list of examples why simple bottom up formatting is not a good approach, and why we should do a refactoring.

### Additions
- tests for validating example files
- script for updating example file output

### CODE REVIEW

File: examples/gmt-building-out.mo [hunk 1]
```diff
@@ -88,78 +88,78 @@ model building
...
-  connect(disFloHea.QActTot_flow, QHea_flow)
-    annotation (Line(points={{-119,-96}, {223.5,-96}, {223.5, 280}, {320, 280}}, color={0, 0, 127}));
-  connect(disFloCoo.QActTot_flow, QCoo_flow)
-    annotation (Line(points={{-119,-156}, {230,-156}, {230, 240}, {320, 240}}, color={0, 0, 127}));
-  connect(maxTSet.y, terUni.TSetCoo)
-    annotation (Line(points={{-268, 200}, {-240, 200}, {-240,-46.6667}, {-200.833,-46.6667}}, color={0, 0, 127}));
-  connect(minTSet.y, terUni.TSetHea)
-    annotation (Line(points={{-268, 240}, {-220, 240}, {-220,-45}, {-200.833,-45}}, color={0, 0, 127}));
+  connect(disFloHea.port_b,secHeaRet[1])
+    annotation (Line(points={{140,-70},{240,-70},{240,32},{300,32}},color={0,127,255}));
```
Comment (macintoshpie) [inline examples/gmt-building-out.mo:94]: @nllong  see here where the no-space-comma might be undesirable. We get `,color=...` rather than `, color=...`

Comment (nllong) [reply examples/gmt-building-out.mo:94]: strange... but let's leave it for now.

File: examples/gmt-building-out.mo [hunk 2]
```diff
@@ -88,78 +88,78 @@ model building
     nPorts_a1=nZon,
     nPorts_b1=nZon)
     "Chilled water distribution system"
-    annotation (Placement(transformation(extent={{-140,-160}, {-120,-140}})));
+    annotation (Placement(transformation(extent={{-140,-160},{-120,-140}})));
```
Comment (nllong) [inline examples/gmt-building-out.mo:91]: I like the more concise {x1,y1} values without spaces

File: thirdparty/Modelica.g4 [hunk 3]
```diff
@@ -358,7 +358,17 @@ primary
 
 // added to original file
 vector
-   : '{' function_arguments '}'
+   : '{' (array_arguments | array_iterator_constructor) '}'
+   ;
+
+// added to original file
```
Comment (nllong) [inline thirdparty/Modelica.g4:364]: thanks for noting that these are changes from the original file.

**Review by nllong — APPROVED** [review]
looks good.

**Pull request merged**